### PR TITLE
Add flag for immunity to skin irritants

### PR DESCRIPTION
**Pull request by zachary-kaelan**
#### Summary

Infrastructure "Add IRRITANT_IMMUNE flag for describing immunity to skin irritants"

#### Purpose of change

When setting up an effect condition, I noted a significant number of potential mutations and mutation armors, and a few different armor suits, that could all reasonably provide total protection of the skin from irritating substances. The list in the effect condition was getting really long.

#### Describe the solution

Create a new flag, `IRRITANT_IMMUNE`, and add it to anything that seems like it would either prevent irritating substances from making contact with the skin. or just remove the skin entirely. I based it on coverage, existing flags, and descriptions.

#### Describe alternatives you've considered

Use the `ACID_IMMUNE` flag instead. While a heavy coat of stiff fur would reasonably prevent a spray of itch powder from making contact with the skin, acid would burn straight through.

Use the `WATERPROOF` and/or `RAINPROOF` flags instead. While letting a few droplets of water slip between your glove and arm sleeves isn't worth fussing about (it will just evaporate), any amount of itching powder slipping through would _suck_, and it would find its way over a lot of the rest of your body as you're swinging weapons and ducking and weaving and such.

#### Testing

The JSON style checker will yell at me if something is goofed.

#### Additional context

This is just for "all-or-nothing" things that cover basically all the skin on the body. More nuanced calculations will presumably come in #72271.

### CODE REVIEW

File: data/json/bionics.json [hunk 1]
```diff
@@ -589,7 +589,7 @@
       [ "hand_r", 1 ],
       [ "mouth", 1 ]
     ],
-    "flags": [ "BIONIC_NPC_USABLE", "BIONIC_SHOCKPROOF" ],
+    "flags": [ "BIONIC_NPC_USABLE", "BIONIC_SHOCKPROOF", "IRRITANT_IMMUNE" ],
```
Comment (Maleclypse) [inline data/json/bionics.json:592]: Under the skin won’t be irritant immune. I’m not even sure it should be shockproof

Comment (zachary-kaelan) [reply data/json/bionics.json:592]: I can't remember what I thought process behind that was. But I believe the `BIONIC_SHOCKPROOF` flag just means if you get electrocuted it won't get messed up.

File: data/json/mutations/mutations.json [hunk 2]
```diff
@@ -8039,7 +8045,7 @@
     "category": [ "PLANT" ],
     "scent_type": "sc_flower",
     "encumbrance_covered": [ [ "foot_l", 10 ], [ "foot_r", 10 ] ],
-    "flags": [ "TOUGH_FEET", "ROOTS3" ]
+    "flags": [ "TOUGH_FEET", "ROOTS3", "IRRITANT_IMMUNE" ]
```
Comment (Maleclypse) [inline data/json/mutations/mutations.json:8048]:
```suggestion
    "flags": [ "TOUGH_FEET", "ROOTS3" ]
```
Chlorophyll doesn't make plants more or less immune to products targeting it's cells.

File: data/json/mutations/mutations.json [hunk 3]
```diff
@@ -8948,6 +8954,7 @@
     "vitamin_cost": 60,
     "description": "Your flesh is highly poisonous, and creatures that bite you will receive a nasty surprise.",
     "category": [ "FISH", "GASTROPOD" ],
+    "flags": [ "IRRITANT_IMMUNE" ],
```
Comment (Maleclypse) [inline data/json/mutations/mutations.json:8957]:
```suggestion
```

File: data/json/mutations/mutations.json (hunk 2 above)
Comment (zachary-kaelan) [reply data/json/mutations/mutations.json]: Yeah, you're right.

File: data/json/mutations/mutations.json (hunk 3 above)
Comment (zachary-kaelan) [reply data/json/mutations/mutations.json]: I'm not sure why I put it for that one, it wouldn't do anything against chemicals.

**Comment by Maleclypse** [comment]
OK I still have some questions about some of these like patch skin and lupine fur but overall I think this makes sense.

**Pull request merged**